Library/Homebrew/cask/lib/hbc/artifact/zap.rb

```diff
@@ -3,17 +3,8 @@
 module Hbc
   module Artifact
     class Zap < UninstallBase
-      def install_phase
-        odebug "Nothing to do. The zap artifact has no install phase."
-      end
-
       def uninstall_phase
-        odebug "Nothing to do. The zap artifact has no uninstall phase."
-      end
-
```

Comment:
It looks like this method has a function though it's overriding as an empty. Without this, uninstall commands function like zap commands because it looks for the Artifacts containing uninstall_phase [installer.rb: ... Artifact.for_cask( ... .respond_to?(:uninstall_phase) ...] then executes them including the zap artifact. When called via uninstall command, this artifact should do nothing as before. Zap function inside the installer.rb file should also call zap_phase instead of uninstall_phase.